### PR TITLE
Delete all keybindings of view

### PR DESCRIPTION
**Pull request by govlas**

### CODE REVIEW

File: gui.go [hunk 1]
```diff
@@ -246,6 +246,19 @@ func (g *Gui) DeleteKeybinding(viewname string, key interface{}, mod Modifier) e
 	return errors.New("keybinding not found")
 }
 
+// DeleteKeybindings deletes all keybindings of view
+func (g *Gui) DeleteKeybindings(viewname string) {
+	i := 0
+	for i < len(g.keybindings) {
```
Comment (jroimartin) [inline gui.go]: What about creating a new slice instead of modifying the one you are iterating? It would probably make the code clearer.

Comment (govlas) [inline gui.go]: New slice  allocates a new block of memory that is not good for performance. 
But we can reuse old slice:
```go
	res := g.keybindings[:0]
	for _, kb := range g.keybindings {
		if kb.viewName != viewname {
			res = append(res, kb)
		}
	}
	g.keybindings = res
```

Comment (jroimartin) [inline gui.go]: That's true. However, in this specific case the performance is not that critical. This is not an operation you want to execute continuously, that's why I prefer to put code clarity first.

Comment (govlas) [inline gui.go]: What do you think about code in my previous comment?
The code is reasonably clean in my opinion.

Comment (jroimartin) [inline gui.go]: To be honest, in this case I don't see the point of reusing the slice.

**Comment by jroimartin** [comment]
Merged!

**Pull request closed without merging**